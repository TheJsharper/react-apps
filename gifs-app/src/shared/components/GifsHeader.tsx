export interface GifsHeaderProps {
    title: string;
    description: string;
}
const GifsHeader = ({ title, description }: GifsHeaderProps) => {
    return (
        <div className="content-center">
            <h1>{title}</h1>
            <p>{description}</p>
        </div>
    );
};

export default GifsHeader;