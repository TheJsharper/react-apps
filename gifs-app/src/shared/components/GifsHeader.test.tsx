import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import GifsHeader from "./GifsHeader";

describe("GifsHeader", () => {
    it("should render correctly", () => {

        const { container } = render(<GifsHeader title="Sample Title" description="Sample Description" />);

        expect(container).toMatchSnapshot();
    });
    it("should display the correct title and description", () => {
        const title = "Test Title";

        const description = "Test Description";

        const { container } = render(<GifsHeader title={title} description={description} />);

        expect(container).toMatchSnapshot();

        expect(container.querySelector("h1")?.textContent).toBe(title);

        expect(container.querySelector("p")?.textContent).toBe(description);
    });

    it("should match snapshot with different props title, description", () => {
        const title = "Another Title";
        const description = "Another Description";

        const { container } = render(<GifsHeader title={title} description={description} />);

        expect(container).toMatchSnapshot();

        expect(screen.getByText(title)).toBeDefined();

        expect(screen.getByText(description)).toBeDefined();
    });

    it("should handle empty title and description", () => {
        const title = "";
        const description = "";

        const { container } = render(<GifsHeader title={title} description={description} />);
        expect(container).toMatchSnapshot();

        expect(screen.getByRole("heading")).toBeDefined();

        expect(screen.getByRole("paragraph")).toBeDefined();

        expect(screen.getByRole("heading").textContent).toBe(title);

        expect(screen.getByRole("paragraph").textContent).toBe(description);
    });

    it("should handle long title and description", () => {
        const title = "This is a very long title that should be tested for rendering properly in the GifsHeader component";
        const description = "This is a very long description that should also be tested to ensure that it renders correctly and does not break the layout of the GifsHeader component in any way.";

        const { container } = render(<GifsHeader title={title} description={description} />);

        expect(container).toMatchSnapshot();

        expect(screen.getByText(title)).toBeDefined();

        expect(screen.getByText(description)).toBeDefined();
    });

    it("should handle special characters in title and description", () => {
        const title = "Title with special characters !@#$%^&*()_+";

        const description = "Description with special characters <>?:\"{}|~`";

        const { container } = render(<GifsHeader title={title} description={description} />);

        expect(container).toMatchSnapshot();

        expect(screen.getByText(title)).toBeDefined();

        expect(screen.getByText(description)).toBeDefined();
    });

    it("should handle numeric title and description", () => {
        const title = "1234567890";

        const description = "0987654321";

        const { container } = render(<GifsHeader title={title} description={description} />);

        expect(container).toMatchSnapshot();

        expect(screen.getByText(title)).toBeDefined();

        expect(screen.getByText(description)).toBeDefined();
    });


    it("should handle title and description with selector querySelector", () => {
        const title = "Title with leading and trailing whitespace";
        const description = "Description with leading and trailing whitespace";
        const { container } = render(<GifsHeader title={title} description={description} />);

        const divElement = container.querySelector(".content-center");

        expect(divElement).toBeDefined();

        expect(screen.getByText(title)).toBeDefined();

        expect(screen.getByText(description)).toBeDefined();
    });

    it("should handle title and description with different casing", () => {
        const title = "tItLe wItH diFfErEnT cAsInG";
        const description = "dEsCrIpTiOn wItH diFfErEnT cAsInG";
        render(<GifsHeader title={title} description={description} />);

        expect(screen.getByText(title)).toBeDefined();
        expect(screen.getByText(description)).toBeDefined();
    });


    it("should handle title and description with emojis", () => {
        const title = "Title with emojis 😊🚀🎉";
        const description = "Description with emojis 👍💡🔥";
        const { container } = render(<GifsHeader title={title} description={description} />);
        expect(container).toMatchSnapshot();

        expect(screen.getByText(title)).toBeDefined();
        expect(screen.getByText(description)).toBeDefined();

    })

    it("should handle title and description with HTML tags", () => {
        const title = "Bold Title";
        const description = "Italic Description";

        const { container } = render(<GifsHeader title={title} description={description} />);

        const div = container.querySelector(".content-center");

        const h1 = container.querySelector("h1");

        const p = container.querySelector("p");

        expect(div).toBeDefined();

        expect(h1).toBeDefined();

        expect(p).toBeDefined();


        expect(h1?.innerHTML).toBe(title);

        expect(p?.innerHTML).toBe(description);
    });


});
