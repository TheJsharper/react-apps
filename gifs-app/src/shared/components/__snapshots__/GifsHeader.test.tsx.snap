// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GifsHeader > should display the correct title and description 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
      Test Title
    </h1>
    <p>
      Test Description
    </p>
  </div>
</div>
`;

exports[`GifsHeader > should handle empty title and description 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1 />
    <p />
  </div>
</div>
`;

exports[`GifsHeader > should handle long title and description 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
      This is a very long title that should be tested for rendering properly in the GifsHeader component
    </h1>
    <p>
      This is a very long description that should also be tested to ensure that it renders correctly and does not break the layout of the GifsHeader component in any way.
    </p>
  </div>
</div>
`;

exports[`GifsHeader > should handle numeric title and description 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
      1234567890
    </h1>
    <p>
      0987654321
    </p>
  </div>
</div>
`;

exports[`GifsHeader > should handle special characters in title and description 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
      Title with special characters !@#$%^&*()_+
    </h1>
    <p>
      Description with special characters &lt;&gt;?:"{}|~\`
    </p>
  </div>
</div>
`;

exports[`GifsHeader > should handle title and description with HTML tags 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
      &lt;b&gt;Bold Title&lt;/b&gt;
    </h1>
    <p>
      &lt;i&gt;Italic Description&lt;/i&gt;
    </p>
  </div>
</div>
`;

exports[`GifsHeader > should handle title and description with different casing 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
      tItLe wItH diFfErEnT cAsInG
    </h1>
    <p>
      dEsCrIpTiOn wItH diFfErEnT cAsInG
    </p>
  </div>
</div>
`;

exports[`GifsHeader > should handle title and description with emojis 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
      Title with emojis üòäüöÄüéâ
    </h1>
    <p>
      Description with emojis üëçüí°üî•
    </p>
  </div>
</div>
`;

exports[`GifsHeader > should handle title and description with whitespace 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
         Title with leading and trailing whitespace   
    </h1>
    <p>
         Description with leading and trailing whitespace   
    </p>
  </div>
</div>
`;

exports[`GifsHeader > should handle title and description with whitespace 2`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
         Title with leading and trailing whitespace   
    </h1>
    <p>
         Description with leading and trailing whitespace   
    </p>
  </div>
</div>
`;

exports[`GifsHeader > should match snapshot with different props 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
      Another Title
    </h1>
    <p>
      Another Description
    </p>
  </div>
</div>
`;

exports[`GifsHeader > should match snapshot with different props title, description 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
      Another Title
    </h1>
    <p>
      Another Description
    </p>
  </div>
</div>
`;

exports[`GifsHeader > should render correctly 1`] = `
<div>
  <div
    class="content-center"
  >
    <h1>
      Sample Title
    </h1>
    <p>
      Sample Description
    </p>
  </div>
</div>
`;
